src/Server/RestRouterHandlers/SpecDocHandler.cpp

```diff
@@ -0,0 +1,79 @@
+#include "SpecDocHandler.h"
+
+#include <Common/Config/YAMLParser.h>
+
+#include <Poco/DirectoryIterator.h>
+#include <Poco/Path.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int RESOURCE_NOT_FOUND;
+}
+
+namespace
+{
+String getSpecPath(const String & config_path)
+{
+    String spec_path = "";
+    Poco::Path path(config_path);
+
+    if (!path.parent().toString().empty() && Poco::File(path.parent().toString() + "/clickhouse-spec").exists())
```

Comment:
It is better to use append here to join the paths

Reply:
fixed